walkthroughs/howto-grpc-ingress-gateway/README.md

```diff
@@ -0,0 +1,107 @@
+## Overview
+
+This example shows how we can route between gRPC ingress gateway and servers using App Mesh.
+
+![System Diagram](./howto-grpc-ingress-gateway.png "System Diagram")
+
+### Color Server
+
+The Color Server is a gRPC server that implements `SetColor` and `GetColor` methods from [color.ColorService](./color.proto). Additionally, it implements the [gRPC Health Checking Protocol](https://github.com/grpc/grpc/blob/master/doc/health-checking.md) which we will configure App Mesh to use as the health check for the its virtual-nodes.
+
+### Color Client
+
+The front-end is Virtual Gateway that maintains a persistent gRPC connection to the Color Server. The Virtual Gateway will be connected to an internet-facing NLB. It forwards the gRPC connections to the same methods in [color.ColorService](./color.proto).
+
+## Setup
+
+1. Clone this repository and navigate to the walkthroughs/howto-grpc-ingress-gateway folder, all commands will be ran from this location.
+    ```
+    cd walkthroughs/howto-grpc-ingress-gateway
+    ```
+2. **Project Name** used to isolate resources created in this demo from other's in your account. e.g. howto-grpc-ingress-gateway
+    ```
+    export PROJECT_NAME=howto-grpc-ingress-gateway
+    ```
+3. **Your** account id:
+    ```
+    export AWS_ACCOUNT_ID=<your_account_id>
+    ```
+4. **Region** e.g. us-west-2
+    ```
+    export AWS_DEFAULT_REGION=us-west-2
+    ```
+5. **ENVOY_IMAGE** set to the location of the App Mesh Envoy container image, see https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+    ```
+    export ENVOY_IMAGE=...
+    ```
+6. **KEY_PAIR** set to the name of an EC2 key pair. We will use this key pair to access a bastion host in the generated VPC to look at the stats collected by the Envoy proxy. See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-key-pairs.html
```

Comment:
We don't use the bastion in this example.

Reply:
I used the bastion extensively to debug and troubleshoot the issues. I think it's very useful.

Reply:
fine by me. I knew that :P